homeassistant/helpers/entity_registry.py

```diff
@@ -277,19 +280,28 @@ def async_clear_config_entry(self, config_entry):
 
 
 @bind_hass
-async def async_get_registry(hass) -> EntityRegistry:
+async def async_get_registry(hass: HomeAssistantType) -> EntityRegistry:
     """Return entity registry instance."""
-    task = hass.data.get(DATA_REGISTRY)
+    reg_or_evt = hass.data.get(DATA_REGISTRY)
+
+    if not reg_or_evt:
+        evt = hass.data[DATA_REGISTRY] = Event()
+
+        reg = EntityRegistry(hass)
+        await reg.async_load()
 
-    if task is None:
-        async def _load_reg():
-            registry = EntityRegistry(hass)
-            await registry.async_load()
-            return registry
+        hass.data[DATA_REGISTRY] = reg
+        evt.set()
+        return reg
 
-        task = hass.data[DATA_REGISTRY] = hass.async_create_task(_load_reg())
+    if isinstance(reg_or_evt, Event):
+        # This shouldn't be possible in production (has been tested)
```

Comment:
Let's drop these comments. We should not write comments that rely on implementation details of other parts of the system that are today interacting with this. It makes it sound like it could be considered to be removed, but it cannot.